modules/gltf/src/lib/extensions/EXT_structural_metadata.ts

```diff
@@ -467,65 +514,67 @@ function getPropertyDataNumeric(
     return [];
   }
 
-  // Single value (not an array)
-  const attributeValueArray: number[] = [];
-  for (let index = 0; index < numberOfElements; index++) {
-    const value = valuesData[index];
-    attributeValueArray.push(value);
-  }
-  return attributeValueArray;
+  return valuesData;
 }
 
+/**
+ * Parse variable-length array data
+ * @param valuesData - values in a flat typed array
+ * @param numberOfElements - number of rows in the property table
+ * @param arrayOffsets - offsets of nested arrays in the flat values array
+ * @param valuesDataBytesLength - data byte length
+ * @param valueSize - value size in bytes
+ * @returns array of typed arrays
+ */
 function handleVariableLengthArrayNumeric(
-  valuesData: TypedArray,
+  valuesData: BigTypedArray,
   numberOfElements: number,
   arrayOffsets: TypedArray,
   valuesDataBytesLength: number,
-  elementSize: number
-) {
-  const attributeValueArray: number[][] = [];
+  valueSize: number
+): BigTypedArray[] {
+  const attributeValueArray: BigTypedArray[] = [];
   for (let index = 0; index < numberOfElements; index++) {
-    const array: number[] = [];
     const arrayOffset = arrayOffsets[index];
     const arrayByteSize = arrayOffsets[index + 1] - arrayOffsets[index];
-    if (arrayByteSize + arrayOffset <= valuesDataBytesLength) {
-      const typedArrayOffset = arrayOffset / elementSize;
-      const elementCount = arrayByteSize / elementSize;
-      for (let i = 0; i < elementCount; i++) {
-        const value = valuesData[typedArrayOffset + i];
-        array.push(value);
-      }
+    if (arrayByteSize + arrayOffset > valuesDataBytesLength) {
+      break;
     }
-    attributeValueArray.push(array);
+    const typedArrayOffset = arrayOffset / valueSize;
+    const elementCount = arrayByteSize / valueSize;
+    attributeValueArray.push(valuesData.slice(typedArrayOffset, typedArrayOffset + elementCount));
   }
   return attributeValueArray;
 }
 
+/**
+ * Parse fixed-length array data
+ * @param valuesData - values in a flat typed array
+ * @param numberOfElements - number of rows in the property table
+ * @param arrayCount - nested arrays length
+ * @returns array of typed arrays
+ */
```

Comment:
Naming: Why `handle` instead of `parse` makes it sounds like it reacts to an event or something?

Reply:
Done